File: nexus-reader/src/api/voice.ts
import { $get, $post, $delete } from './client'

export interface VoiceModelMetadata {
    id: string
    name: string
    type: string
    metadata: Record<string, string>
    modelSize: number
    sampleDuration: number
    createdAt: number
    updatedAt: number
}

export const voiceApi = {
    // Model Metadata
    getMetadata: () => $get<VoiceModelMetadata[]>('/voice/metadata'),
    saveMetadata: (model: VoiceModelMetadata) => $post<void>('/voice/metadata', model),
    deleteMetadata: (id: string) => $delete<void>(`/voice/metadata/${id}`),

    // Configuration
    getConfig: (key: string) => $get<string | null>(`/voice/config/${key}`),
    saveConfig: (key: string, value: string) => $post<void>(`/voice/config/${key}`, { value }),
}

File: api/nexus-core/src/types/voice.rs
use serde::{Deserialize, Serialize};
use std::collections::HashMap;

/// Voice model metadata for TTS
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct VoiceModelMetadata {
    pub id: String,
    pub name: String,
    pub r#type: String,
    pub metadata: HashMap<String, String>,
    pub model_size: u64,
    pub sample_duration: f32,
    pub created_at: i64,
    pub updated_at: i64,
}
